app/controllers/oidc_controller.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# OidcController is used for performing an OpenID Connect auth
+# flow against Dex, allowing Velum to generate a complete kubeconfig
+# file for a user, including their JWT token
+class OidcController < ApplicationController
+  # allow anyone to start this, the auth will happen inside
+  # Dex.
+  skip_before_action :authenticate_user!  
+
+  def start
```

Comment:
i'd prefer if `caasp-cli` would be able to connect to this endpoint, then we have a single authorization entrypoint for UI and CLI